File: src/main/java/com/lanka_guide/tripplanner/model/activity/Visit.java
package com.lanka_guide.tripplanner.model.activity;

import com.lanka_guide.tripplanner.model.place.Place;

import java.time.ZonedDateTime;

public class Visit extends Activity {
    Place place;
    public Visit(Place place, ZonedDateTime from, ZonedDateTime to) {
        super(from, to);
        this.place = place;
    }
}

File: src/main/kotlin/com/lankaguide/tripplanner/model/activity/Visit.kt
package com.lankaguide.tripplanner.model.activity

import com.lankaguide.tripplanner.model.place.Place
import java.time.Duration

import java.time.ZonedDateTime

data class Visit(val place: Place, override val start: ZonedDateTime, override val end: ZonedDateTime) : Activity(start, end) {
    constructor(place: Place, start: ZonedDateTime, duration: Duration) : this(place, start, start.plus(duration))
    constructor(place: Place, duration: Duration, end: ZonedDateTime) : this(place, end.minus(duration), end)
}
